### PR TITLE
DataFormats dependency on Geometry Cleanup

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Restructure **Geometry/TrackerGeometryBuilder** package to remove unwanted dependency 
* Allow **DataFormats** dependency only on **Geometry/CommonDetUnit**, **Geometry/CommonTopology** and **Geometry/CaloGeometry**
* Merge RPC analyser code into an RPC DetId unit test
* Move CSC analyser dependent on CSCGeometry to CSC Geometry package
* Move GEM analyser to GEM Geometry package

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
please abort test

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

as expected, there are no differences found in the comparisons

**Comment by jfernan2** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by alja** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by pohsun** [comment]
+1

**Review by perrotta — COMMENTED** [review]
Is the configuration file to run CSCDetIdAnalyzer (previously in DataFormats/MuonDetId/test/cdan_cfg.py) moved somewhere else? The analyzer itself is now moved into CSCGeometry/test/stubs/CSCDetIdAnalyzer.cc, but I miss how it is run...

File: Geometry/CommonDetUnit/src/PixelGeomDetType.cc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// #include "Geometry/CommonDetUnit/interface/PixelGeomDetType.h"
```
[the remark this answers is not in the document]
Comment (ianna) [reply Geometry/CommonDetUnit/src/PixelGeomDetType.cc]: well spotted, thanks! There is definitely room for further cleanup. Would it be ok to do it as a subsequent PR or you'd prefer me to fix it in this one?

[the remark this answers is not in the document]
Comment (ianna) [reply Geometry/CommonDetUnit/src/PixelGeomDetType.cc]: The configuration was not functional hence removed. I plan to add it in a later PR - including a script to run all tests.

**Comment by perrotta** [comment]
+1
- Rather technical restructuring of Geometry dependent includes
- Jenkins tests pass and show no effect on reco outputs, as it should

**Comment by fabiocos** [comment]
@ggovi @civanch @sbein @ssekmen @rekovic the changes in your domains look a simple change of include paths, I will move forward with the integration of this big PR tonight

@ianna I understand that you will have a follow-up cleaning some code (see the comment of @perrotta  that I share)

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**